plugins/c8y_log_plugin/src/main.rs

```diff
@@ -154,6 +126,72 @@ async fn run(
     }
 }
 
+pub async fn process_mqtt_messages(
+    message: Option<Message>,
+    plugin_config: &LogPluginConfig,
+    mqtt_client: &mut Connection,
+    http_client: &mut JwtAuthHttpProxy,
+    config_file: &Path,
+) -> Result<(), anyhow::Error> {
+    let response_topic_health = Topic::new_unchecked("tedge/health/c8y-log-plugin");
+    if let Some(message) = message {
+        if is_c8y_bridge_up(&message) {
+            let plugin_config = read_log_config(config_file);
+            let () = handle_dynamic_log_type_update(&plugin_config, mqtt_client).await?;
+        }
+        match message.topic.name.as_str() {
+            "tedge/health-check" | "tedge/health-check/c8y-log-plugin" => {
+                let health_status = json!({
+                    "status": "up",
+                    "pid": process::id(),
+                    "time": OffsetDateTime::now_utc().unix_timestamp(),
+                })
+                .to_string();
+
+                let health_message = Message::new(&response_topic_health, health_status);
```

Comment:
This should be a function shared by all thin-edge components and not copy-pasted.

Reply:
fixed